src/sage/rings/padics/witt_vector.py

```diff
@@ -0,0 +1,914 @@
+                ).vec()
+            else:
+                vec_k = (-x for x in vec_k)
+
+        self._vec = tuple([R(x) for x in vec_k])
+
+    def _invert_(self):
+        """
+        Return the inverse of ``self``.
+
+        EXAMPLES::
+
+            sage: W = WittVectorRing(GF(3), prec=4)
+            sage: t = W([1,2,0,1])
+            sage: ~t
+            (1, 1, 1, 0)
+        """
+        if not self[0].is_unit():
+            raise ZeroDivisionError(f"Inverse of {self} does not exist.")
+        P = self.parent()
+
+        if self == P.one():
+            return self
+        if self._prec == 1:
+            return P((self[0]**-1, ))
+
+        # Strategy: Multiply ``self`` by ``(Y_0, Y_1, ...)``, set equal
+        # to (1, 0, 0, ...), and solve.
+        var_names = [f'Y{i}' for i in range(1, self._prec)]
+        poly_ring = PolynomialRing(P.base(), var_names)
+        inv_vec = list((self[0]**-1,) + poly_ring.gens())
+        # We'll fill this in one-by-one
+
+        from sage.rings.padics.witt_vector_ring import WittVectorRing
+        W = WittVectorRing(poly_ring, p=P.prime(), prec=self._prec)
+        prod_vec = (W(self._vec) * W(inv_vec)).vec()
+        for i in range(1, self._prec):
+            poly = prod_vec[i](inv_vec[1:])
+            Y_i = poly.parent().gens()[i - 1]
+            try:
+                inv_vec[i] = (-poly.constant_coefficient()
+                              / poly.monomial_coefficient(Y_i))
+            except ZeroDivisionError:
+                raise ZeroDivisionError(f"Inverse of {self} does not exist.")
+            try:
+                inv_vec[i] = P.base()(inv_vec[i])
+            except ValueError:
+                raise ZeroDivisionError(f"Inverse of {self} does not exist.")
+
+        return P(inv_vec)
+
+    def _latex_(self):
+        r"""
+        Return a `\LaTeX` representation of ``self``.
+
+        EXAMPLES::
+
+            sage: W=WittVectorRing(ZZ, p=7, prec=3)
+            sage: t=W([6,1,6])
+            sage: latex(t)
+            \left(6, 1, 6\right)
+        """
+        return tuple_function(self._vec)
+
+    def _neg_(self):
+        """
+        Return the opposite of ``self``.
+
+        EXAMPLES::
+
+            sage: W = WittVectorRing(GF(3), prec=4)
+            sage: t = W([1,2,0,1])
+            sage: -t
+            (2, 1, 0, 2)
+        """
+        P = self.parent()
+        # If p == 2, -1 == (-1, -1, -1, ...)
+        # Otherwise, -1 == (-1, 0, 0, ...)
+        if P.prime() == 2:
+            all_ones = P(tuple(-1 for _ in range(self._prec)))
+            return all_ones * self
+        neg_vec = tuple(-self[i] for i in range(self._prec))
+        return P(neg_vec)
+
+    def _repr_(self) -> str:
+        """
+        Return a string representation.
+
+        EXAMPLES::
+
+            sage: W = WittVectorRing(ZZ, p=3, prec=4)
+            sage: t = W([1,2,0,1]); t
+            (1, 2, 0, 1)
+        """
+        return '(' + ', '.join(map(str, self._vec)) + ')'
```

Comment:
```suggestion
        return str(self._vec)
```
Is it something I'm missing?

Reply:
Yes, this adds an unwanted extra comma at the end of the truncated Witt vector representation.